File: Day_10/144. Binary Tree Preorder Traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> preorderTraversal(TreeNode* root) {
        vector<int> t;
        preOrder(root, t);
        return t;
        
    }
    void preOrder(TreeNode* root, vector<int>&p){
        if(root==NULL)
            return;
        p.push_back(root->val);
        preOrder(root->left, p);
        preOrder(root->right, p);
    }
};

File: Day_10/144. Binary Tree Preorder Traversal.js
var preorderTraversal = function(root) {
    const tree = [];
    preorder(root);
    return tree;

    function preorder(root){
        if(root===null)
            return tree;
        tree.push(root.val);
        preorder(root.left);
        preorder(root.right);
    }
};
